terraform/implementation/setup/iam.tf

```diff
@@ -0,0 +1,119 @@
+data "aws_caller_identity" "current" {}
+
+# # create a role that can be assumed to pull and push docker images from 
+data "aws_iam_policy_document" "github_assume_role" {
+  statement {
+    principals {
+      type        = "Federated"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"]
+    }
+    actions = [
+      "sts:AssumeRoleWithWebIdentity"
+    ]
+    condition {
+      test     = "StringEquals"
+      variable = "token.actions.githubusercontent.com:aud"
+      values   = ["sts.amazonaws.com", ]
+    }
+    condition {
+      test     = "StringLike"
+      variable = "token.actions.githubusercontent.com:sub"
+      values = [
+        "repo:${var.oidc_github_repo}:*",
+      ]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "github" {
+  statement {
+    actions = [
+      "appmesh:*",
+      "dynamodb:*",
+      "ec2:*",
+      "ecr:*",
+      "ecs:*",
+      "elasticloadbalancing:*",
+      "iam:*",
+      "logs:*",
+      "s3:*",
+      "servicediscovery:*",
+      "ecs:UpdateService",
+      # "appmesh:DescribeMesh",
```

Comment:
Do you want to put a `#TODO:` note here stating that iam policies will be refactored to least necessary privileges?